crates/types/src/traits/block_contents.rs

```diff
@@ -63,6 +49,8 @@ pub trait BlockPayload:
 pub trait Transaction:
     Clone + Serialize + DeserializeOwned + Debug + PartialEq + Eq + Sync + Send + Committable + Hash
 {
+    /// Get the transaction bytes.
+    fn bytes(&self) -> Vec<u8>;
```

Comment:
I think `bytes` here isn't the same as serialization, since it's getting the inner vector of `Transaction` (i.e., `self.0` for `VIDTransaction`), rather than serializing the whole `Transaction`.

Maybe `inner` or `inner_bytes` is a better name to avoid the confusion.

Reply:
> I think `bytes` here isn't the same as serialization, since it's getting the inner vector of `Transaction` (i.e., `self.0` for `VIDTransaction`), rather than serializing the whole `Transaction`.

What's the difference? A transaction is just a vec<u8> so wouldn't its serialization be identical to self.0?

Do we have reason to require two different ways to get the bytes of a tx?

Reply:
Agreed. Actually, this function is only used to chain txn bytes for VID dispersal computation, which IIUC will be unnecessary once jellyfish supports passing in a vec of transactions. But anyway, I think it makes sense to choose either `self.0` or `serde`, not both.

Reply:
Jellyfish will take an iterator over bites, but you still need bites

Reply:
An alternative is to impl AsRef<u8>